polaris-react/src/components/Scrollable/Scrollable.tsx

```diff
@@ -36,218 +32,86 @@ export interface ScrollableProps extends React.HTMLProps<HTMLDivElement> {
   onScrolledToBottom?(): void;
 }
 
-interface State {
-  topShadow: boolean;
-  bottomShadow: boolean;
-  scrollPosition: number;
-  canScroll: boolean;
-}
-
-export class Scrollable extends Component<ScrollableProps, State> {
-  static ScrollTo = ScrollTo;
-  static forNode(node: HTMLElement): HTMLElement | Document {
-    const closestElement = node.closest(scrollable.selector);
-    return closestElement instanceof HTMLElement ? closestElement : document;
-  }
-
-  state: State = {
-    topShadow: false,
-    bottomShadow: false,
-    scrollPosition: 0,
-    canScroll: false,
-  };
-
-  private stickyManager = new StickyManager();
-
-  private scrollArea: HTMLElement | null = null;
-
-  private handleResize = debounce(
-    () => {
-      this.handleScroll();
-    },
-    50,
-    {trailing: true},
-  );
-
-  componentDidMount() {
-    if (this.scrollArea == null) {
-      return;
-    }
-    this.stickyManager.setContainer(this.scrollArea);
-    this.scrollArea.addEventListener('scroll', () => {
-      window.requestAnimationFrame(this.handleScroll);
-    });
-    window.addEventListener('resize', this.handleResize);
-    window.requestAnimationFrame(() => {
-      this.handleScroll();
-      if (this.props.hint) {
-        this.scrollHint();
-      }
-    });
-  }
-
-  componentWillUnmount() {
-    if (this.scrollArea == null) {
-      return;
+export function Scrollable({
+  children,
+  className,
+  horizontal,
+  vertical = true,
+  shadow,
+  hint,
+  focusable,
+  onScrolledToBottom,
+  ...rest
+}: ScrollableProps) {
+  const [topShadow, setTopShadow] = useState(false);
+  const [bottomShadow, setBottomShadow] = useState(false);
+  const stickyManager = useRef(new StickyManager());
+  const scrollArea = useRef<HTMLDivElement>(null);
+  const scrollTo = useCallback((scrollY: number) => {
+    scrollArea.current?.scrollTo({top: scrollY, behavior: 'smooth'});
```

Comment:
Probably out of scope for this PR but I'd love if we could add a check for perfers reduced motion and only use smooth behaviour if that check returns false

If you have time it would be a nice add, I think